src/datasources/db/cached-query-resolver.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { fakeJson } from '@/__tests__/faker';
+import { FakeCacheService } from '@/datasources/cache/__tests__/fake.cache.service';
+import { CachedQueryResolver } from '@/datasources/db/cached-query-resolver';
+import { ILoggingService } from '@/logging/logging.interface';
+import { faker } from '@faker-js/faker';
+import { InternalServerErrorException } from '@nestjs/common';
+import postgres, { MaybeRow } from 'postgres';
+
+const mockLoggingService = jest.mocked({
+  debug: jest.fn(),
+  error: jest.fn(),
+} as jest.MockedObjectDeep<ILoggingService>);
+
+const mockQuery = jest.mocked({
+  execute: jest.fn(),
+} as jest.MockedObjectDeep<postgres.PendingQuery<MaybeRow[]>>);
+
+describe('CachedQueryResolver', () => {
+  let fakeCacheService: FakeCacheService;
+  let target: CachedQueryResolver;
+
+  beforeAll(() => {
+    fakeCacheService = new FakeCacheService();
+    target = new CachedQueryResolver(mockLoggingService, fakeCacheService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+  });
+
+  describe('get', () => {
+    it('should return the content from cache if it exists', async () => {
+      const cacheDir = { key: 'key', field: 'field' };
+      const ttl = faker.number.int({ min: 1, max: 1000 });
+      const value = fakeJson();
+      await fakeCacheService.set(cacheDir, JSON.stringify(value), ttl);
+
+      const actual = await target.get({
+        cacheDir,
+        query: mockQuery,
+        ttl,
+      });
+
+      expect(actual).toBe(value);
+      expect(mockLoggingService.debug).toHaveBeenCalledWith({
+        type: 'cache_hit',
+        key: 'key',
+        field: 'field',
+      });
+      const cacheContent = await fakeCacheService.get(cacheDir);
+      expect(cacheContent).toBe(JSON.stringify(value));
```

Comment:
Nit: this is unecessary as we set the cache value.

Reply:
Indeed. Thanks! Removed in [e4138c0](https://github.com/safe-global/safe-client-gateway/pull/1777/commits/e4138c04c97298f92c2851b6c259cfd9e112545b)